src/query/storage/m3/storage.go

```diff
@@ -59,11 +60,13 @@ type m3storage struct {
 
 // NewStorage creates a new local m3storage instance.
 func NewStorage(
+	tagOptions models.TagOptions,
```

Comment:
nit: Move to last arg as this is an options arg and none others are?

Reply:
sgtm